src/api/app/models/binary_release.rb

```diff
@@ -47,14 +47,15 @@ def self.update_binary_releases_via_json(repository, json, time = Time.now)
                  obsolete_time:  nil,
                  modify_time:    nil }
         # check for existing entry
-        existing = oldlist.where(hash)
-        Rails.logger.info "ERROR: multiple matches, cleaning up: #{existing.inspect}" if existing.count > 1
+        matching_binaries = oldlist.where(hash)
+        Rails.logger.info "ERROR: multiple matches, cleaning up: #{matching_binaries.inspect}" if matching_binaries.exist?
         # double definition means broken DB entries
-        existing.offset(1).destroy_all
+        matching_binaries.offset(1).destroy_all
 
         # compare with existing entry
-        if existing.count == 1
-          entry = existing.first
+        entry = matching_binaries.first
+
+        if entry
```

Comment:
[nitpick] entry.present?

Reply:
`matching_binaries` is an ActiveRecord query result. So `matching_binaries.first` is either nil or a binary release object